### DISCUSSION TITLE
Track Alt Types

### DISCUSSION THREAD

**Discussion by spacek531** [OpenRCT2/OpenRCT2#17583] [Feature requests]
There's been a lot of discussion about alt types, but not a lot of putting down hard requirements.

What are alt track types?

Alt track types are subtypes of a ride type's paint code which are drawn in a different style with no change in behavior (Covered track elements do have a change in behavior). The most easily understood example is the different Intamin tracks:

![image](https://user-images.githubusercontent.com/3454156/180367683-0e34c9d8-c62e-4d37-be4d-e8386adeac1b.png)

Some people have brought up the idea that alt track types can be used for alternate brake fin placements, such as using inline brake fins on the Giga coaster instead of wing-mounted fins.

@X123M3-256 @karst I believe you two both had some ideas on what alt track types would need.

I think this discussion will be valuable for defining how much room alt track types should take in the tile element struct, and how they will be defined in the future track type files.

**Comment by karst** [comment]
I cannot think of a manufacturer other than Intamin that has different track depending on the forces displayed on the track. Correct me if I am wrong. 
In case of intamin, their old style track had three different variants and so does the new style. 

Old Intamin Track
-------------------
![image](https://user-images.githubusercontent.com/1516893/180371708-7dae9e23-2a88-4868-abba-3ff812743fb6.png) 
![image](https://user-images.githubusercontent.com/1516893/180372247-ab065759-a5f4-4b45-8a94-5c87762c6f41.png)

In the case of all Intamin coasters the highest force elements are built with a square shaped spine (what you can currently in game find on the impulse coaster).
On places that have less forces on them you will see a triangle shaped spine.
On near forceless areas you will find spineless track. Think of places like the brake run

New Intamin Track
--------------------
![image](https://user-images.githubusercontent.com/1516893/180372939-53a2e075-adde-4b01-bdef-fd1630d28a7f.png)
![image](https://user-images.githubusercontent.com/1516893/180373308-381af634-0e1d-45be-9870-c288696bc4a0.png)
In the case of the new spine system the square spine has been replaced for a double spined system. 
The triangle has a singular spine, and the flat piece is the same as on the old track. The functionality is the same, looks are different. I don't exactly know what the benefit of the new track is but it's either cheaper to manufacture or easier to shape (smoother ride, higher stats, possibly higher support limit).
 

Use case in the game
-----------------------
Personally I was thinking it might be a fun addition to the game to introduce a roughness system. In the case of Intamin coasters the different kinds of spine could reduce the roughness in a certain forceful spot, mostly with high laterals or positive verticals it would be incentivized to use the double spine/square spine. But it would be more expensive to build it with it. So in places where there's less forces you can use the standard single spine/triangle spine.  
The cheapest option would be the spineless track, but you need to make sure that the forces here are nearly forceless. This would work for an inline twist or the brake run. 
You can choose to cheap out a bit by building with a spine that might not be as good for said forces, but the rating of your coaster will turn out worse for it.

I do not know what @X123M3-256's specific ideas are for the different track styles.

**Comment by spacek531** [comment]
Premier SkyRocket:
![image](https://user-images.githubusercontent.com/3454156/180373700-f0872ac6-5422-47c7-804c-b5b86e26c7e7.png)

Gerstlauer Eurofighter and Infinity:
![image](https://user-images.githubusercontent.com/3454156/180374282-68400a33-9eef-431b-88cf-9b1d36c8b27d.png)

Chance Rides Big Dipper:
![image](https://user-images.githubusercontent.com/3454156/180374638-fdd891be-f762-4855-99fd-480b44ba8dfd.png)

Zamperla Zig Zag Coaster:
![image](https://user-images.githubusercontent.com/3454156/180375574-495bd96f-c3b9-4695-98d8-b37232f9392f.png)

Vekoma Junior Coaster:
![image](https://user-images.githubusercontent.com/3454156/180375908-5aeecb18-ad8b-45bc-a248-019a625eae0c.png)
![image](https://user-images.githubusercontent.com/3454156/180376175-d3122f86-1eb2-4efb-8388-c1fd19188391.png)

SBF Visa Race Coaster:
![image](https://user-images.githubusercontent.com/3454156/180376544-c1fb365b-0562-407b-9e5f-2d621a9bed8e.png)

Intamin Multidimensional Coaster:
![image](https://user-images.githubusercontent.com/3454156/180380898-7f0dfd2f-4689-4aab-ba74-3ac671087dd0.png)

**Comment by J-PIE-314** [comment]
This request is similar to https://github.com/OpenRCT2/OpenRCT2/discussions/18392 and is related to https://github.com/OpenRCT2/OpenRCT2/discussions/17572.

**Comment by DJmant** [comment]
> There's been a lot of discussion about alt types, but not a lot of putting down hard requirements.
> 
> What are alt track types?
> 
> Alt track types are subtypes of a ride type's paint code which are drawn in a different style with no change in behavior (Covered track elements do have a change in behavior). The most easily understood example is the different Intamin tracks:
> 
> ![image](https://user-images.githubusercontent.com/3454156/180367683-0e34c9d8-c62e-4d37-be4d-e8386adeac1b.png)
> 
> Some people have brought up the idea that alt track types can be used for alternate brake fin placements, such as using inline brake fins on the Giga coaster instead of wing-mounted fins.
> 
> @X123M3-256 @karst I believe you two both had some ideas on what alt track types would need.
> 
> I think this discussion will be valuable for defining how much room alt track types should take in the tile element struct, and how they will be defined in the future track type files.
